README.md

```diff
@@ -25,6 +25,9 @@ tag, or opt-in to [use a full Git commit SHA] and Dependabot.
 
 ### Trusted publishing
 
+> [!NOTE] Trusted publishing does not support reusable workflows at this time,
+> please use a username/token instead.
```

Comment:
I don't like the imperative suggestion to use the token instead. I'd rather use an FAQ section like https://github.com/pypa/gh-action-pypi-publish#specifying-a-different-username and link there like it's down a few lines below. Maybe, even extend that paragraph with a sentence like “For example, when you oppose to moving the action call outside a reusable workflow”.

This is because the preferred solution is to use secretless publishing. And since it's already recommended to have it in a separate job, it shouldn't be too bad to have people copy-and-paste a job with 2 steps in it (download the dists built in a separate job + call this action).

Reply:
If you maintain 15 repos that all have the same complex build steps, copy/paste is really obnoxious.. that's why reusable workflows exist. And the download section might be rather large, see https://github.com/robotpy/build-actions/blob/5b0b4e1bfe7bbac2fa04be5ca32f259bcc591627/.github/workflows/package-ci.yml#L293

However, I hadn't realized that you can have jobs that call reusable workflows and jobs that do normal things in the same workflow file, that's useful to know (and I feel like it wasn't originally supported?). I'll update that.

Reply:
> If you maintain 15 repos that all have the same complex build steps, copy/paste is really obnoxious.. that's why reusable workflows exist. And the download section might be rather large, see https://github.com/robotpy/build-actions/blob/5b0b4e1bfe7bbac2fa04be5ca32f259bcc591627/.github/workflows/package-ci.yml#L293

I totally understand that and I'm quite interested in reducing such burden, maintaining even more repos myself. I like the approach outlined @ https://blog.jaraco.com/skeleton/ that's used to manage hundreds of repos. It simply uses `git-merge` across different repos, bringing common configs into downstreams from a common skeleton. It requires a mindset shift, though — for example, reducing merge conflicts by putting things into separate files or keeping downstream-specific changes in a dedicated place (like at the bottom of the share configs).
Maybe, that'd be useful to you.

> However, I hadn't realized that you can have jobs that call reusable workflows and jobs that do normal things in the same workflow file, that's useful to know (and I feel like it wasn't originally supported?). I'll update that.

Pretty sure this was supported since the introduction of the reusable workflows. It's also nice to make use of in-repo reusable workflows for modularization. I suggested this approach in CPython and a few PyPA repos + some of mine, and it does work quite well.

Reply:
@virtuald I looked into your workflows closer and found a few places that could be improved:
- the artifacts can be downloaded in a single step (be careful with the upgrade to v4, though — GH made some breaking changes: https://github.com/actions/upload-artifact/issues/472)
- you still rely on tokens, while you could upgrade to secretless uploads
- you build sdist in the same job as publishing — this is bad, it opens up possibilities for privilege elevation coming from the build machinery
- you build wheels from Git source, but `pip install` builds wheels from an sdist; it's best to build an sdist in one job, download it into the matrix and build wheels from that — it's a good smoke test for knowing that your sdist is in fact usable. `cibuildwheel` can even consume an sdist tarball filepath as an input.